commons-numbers-examples/examples-jmh/src/main/java/org/apache/commons/numbers/examples/jmh/core/StickySumPerformance.java

```diff
@@ -0,0 +1,609 @@
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.Fork;
+import org.openjdk.jmh.annotations.Measurement;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Param;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.Warmup;
+import org.openjdk.jmh.infra.Blackhole;
+
+/**
+ * Executes a benchmark to measure the speed of operations in the {@link LinearCombination} class.
+ * Benchmarks focus on the sticky summation of two double values.
+ *
+ * <p>Details of the sticky bit can be found in:
+ * <blockquote>
+ * Coonen, J.T., "An Implementation Guide to a Proposed Standard for Floating Point
+ * Arithmetic", Computer, Vol. 13, No. 1, Jan. 1980, pp 68-79.
+ * </blockquote>
+ */
+@BenchmarkMode(Mode.AverageTime)
+@OutputTimeUnit(TimeUnit.NANOSECONDS)
+@Warmup(iterations = 5, time = 1, timeUnit = TimeUnit.SECONDS)
+@Measurement(iterations = 5, time = 1, timeUnit = TimeUnit.SECONDS)
+@State(Scope.Benchmark)
+@Fork(value = 1, jvmArgs = {"-server", "-Xms512M", "-Xmx512M"})
+public class StickySumPerformance {
+    /** The mask for the sign bit and the mantissa. */
+    private static final long SIGN_MATISSA_MASK = 0x800f_ffff_ffff_ffffL;
+
+    /** Constant for no method. */
+    private static final String NONE = "none";
+    /** Constant for branched method. */
+    private static final String BRANCHED = "branched";
+    /** Constant for branchless method. */
+    private static final String BRANCHLESS = "branchless";
+    /** Constant for single branch method based on the low bit of the high part. */
+    private static final String BRANCH_ON_HI = "branch_on_hi";
+    /** Constant for single branch method based on the low part. */
+    private static final String BRANCH_ON_LO = "branch_on_lo";
+
+    /**
+     * The factors to sum.
+     */
+    @State(Scope.Benchmark)
+    public static class BiFactors {
+        /** The exponent for small numbers. */
+        private static final long EXP = Double.doubleToRawLongBits(1.0);
+
+        /**
+         * The count of sums.
+         */
+        @Param({"10000"})
+        private int size;
+
+        /**
+         * The fraction of numbers that have a zero round-off.
+         */
+        @Param({"0", "0.1"})
+        private double zeroRoundoff;
+
+        /** Factors. */
+        private double[] a;
+
+        /**
+         * Gets the factors to be summed as pairs. The array length will be even.
+         *
+         * @return Factors.
+         */
+        public double[] getFactors() {
+            return a;
+        }
+
+        /**
+         * Create the factors.
+         */
+        @Setup
+        public void setup() {
+            final UniformRandomProvider rng = RandomSource.create(RandomSource.XO_RO_SHI_RO_1024_PP);
+            a = new double[size * 2];
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The result of `Integer` multiplication is cast into a larger type. If you think the result of the multiplication could exceed the size of an `Integer`, consider casting the arguments of the multiplication to avoid arithmetic overflows. For more information, see [CWE-190](https://cwe.mitre.org/data/definitions/190.html).